File: Frameworks/react-web/tests/conformance/conformance.test.ts
import { describe, it, afterEach } from 'vitest'

import { allScenarios } from '../../../conformance/src/index.ts'
import { ReactDriver } from './react-driver.ts'

// ---------------------------------------------------------------------------
// Conformance runner — each scenario becomes a vitest case. The runner
// skips (via `it.skip`) scenarios whose required capabilities aren't all
// supported by the driver. Any throw inside `scenario.run` becomes a
// test failure.
// ---------------------------------------------------------------------------

describe('ODS conformance (ReactDriver)', () => {
  let driver: ReactDriver | null = null

  afterEach(async () => {
    if (driver) {
      await driver.unmount()
      driver = null
    }
  })

  for (const scenario of allScenarios) {
    const supported = scenario.capabilities.every((c) =>
      new ReactDriver().capabilities.has(c),
    )
    const runner = supported ? it : it.skip

    runner(scenario.name, async () => {
      driver = new ReactDriver()
      await driver.mount(scenario.spec())
      await driver.setSeed(0)
      await driver.setClock('2026-01-01T00:00:00Z')
      await scenario.run(driver)
    })
  }
})

File: Frameworks/flutter-local/test/conformance/conformance_test.dart
/// ODS conformance runner for the Flutter renderer.
///
/// Mirrors conformance.test.ts — iterates the scenarios declared in
/// scenarios.dart and runs each against a fresh FlutterDriver. Scenarios
/// whose required capabilities exceed what the driver advertises are
/// skipped (not failed), matching the TS runner's behavior.
library;

import 'package:flutter_test/flutter_test.dart';

import 'flutter_driver.dart';
import 'scenarios.dart';

void main() {
  group('ODS conformance (FlutterDriver)', () {
    for (final scenario in allScenarios) {
      final driverCaps = FlutterDriver().capabilities;
      final supported = scenario.capabilities.every(driverCaps.contains);

      if (!supported) {
        test(scenario.name, () {}, skip: 'driver lacks required capabilities');
        continue;
      }

      test(scenario.name, () async {
        final driver = FlutterDriver();
        try {
          await driver.mount(scenario.spec());
          await driver.setSeed(0);
          await driver.setClock('2026-01-01T00:00:00Z');
          await scenario.run(driver);
        } finally {
          await driver.unmount();
        }
      });
    }
  });
}
